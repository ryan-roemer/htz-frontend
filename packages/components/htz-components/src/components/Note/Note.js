import React from 'react';
import PropTypes from 'prop-types';
import { createComponent, } from 'react-fela';
import { parseComponentProp, } from '@haaretz/htz-css-tools';
import { responsivePropBaseType, } from '../../propTypes/responsivePropBaseType';
import { textInputVariantType, } from '../TextInput/textInputVariantType';

const TextInputVariants = textInputVariantType;

const propTypes = {
  /** Class(es) to be added to the DOM element.
   * Automatically generated by Fela, do not enter manually.
   */
  className: PropTypes.string,
  /** true if there is a validation error */
  // used for style change
  // eslint-disable-next-line react/no-unused-prop-types
  isError: PropTypes.bool,
  /**
   * Id used to connect the note to input with aria-describedby for a11y reasons,
   * default will generate random id
   */
  noteId: PropTypes.string,
  /** the note text, parent component responsible for changing the text when there is an error */
  text: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.node,
  ]),
  /** The TextInput stylistic variant */
  // used for styling
  // eslint-disable-next-line react/no-unused-prop-types
  variant: PropTypes.oneOfType([
    TextInputVariants,
    PropTypes.arrayOf(
      PropTypes.shape({
        ...responsivePropBaseType,
        value: TextInputVariants.isRequired,
      })
    ),
  ]),
};
const defaultProps = {
  className: null,
  isError: false,
  noteId: null,
  text: null,
  variant: 'primary',
};

const styles = ({ theme, isError, variant, }) => ({
  display: 'block',
  extend: [
    theme.type(-3),
    parseComponentProp(
      undefined,
      variant,
      theme.mq,
      setVariant,
      theme.color,
      isError
    ),
  ],
});

function setVariant(prop, variant, getColor, isError) {
  return {
    color: isError
      ? getColor('input', `${variant}ErrorTextNote`)
      : getColor('input', `${variant}TextNote`),
  };
}

function Note({ className, text, noteId, }) {
  return (
    <span className={className} id={noteId}>
      {text}
    </span>
  );
}

Note.propTypes = propTypes;
Note.defaultProps = defaultProps;

const StyledNote = createComponent(styles, Note, [ 'text', 'noteId', ]);

export default StyledNote;
