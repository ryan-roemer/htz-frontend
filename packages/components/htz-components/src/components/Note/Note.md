### Note

`<Note>` is used to describe an input or display an errorText when the input is error state.

```jsx
<Note text="example text" id="customid" />
```

```jsx
<Note isError text="Error text" id="customid" />
```
