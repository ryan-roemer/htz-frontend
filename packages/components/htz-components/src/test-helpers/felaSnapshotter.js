import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
