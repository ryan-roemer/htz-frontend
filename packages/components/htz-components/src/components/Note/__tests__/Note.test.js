import React from 'react';
import felaSnapshotter from '../../../test-helpers/felaSnapshotter';
import Note from '../Note';

describe('<Note>', () => {
  describe('DOM element', () => {
    it('renders correctly with minimal required props', () => {
      const { component, styles, } = felaSnapshotter(<Note text="text" id="1234" />);
      expect(component).toMatchSnapshot();
      expect(styles).toMatchSnapshot();
    });
    it('renders correctly with `isError prop', () => {
      const { component, styles, } = felaSnapshotter(<Note isError text="error text" id="1234" />);
      expect(component).toMatchSnapshot();
      expect(styles).toMatchSnapshot();
    });
  });
});
