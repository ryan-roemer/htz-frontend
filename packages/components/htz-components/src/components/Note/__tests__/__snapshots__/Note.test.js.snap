// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Note> DOM element renders correctly with \`isError prop 1`] = `
<div>
  <span
    className="htz-a htz-b htz-c htz-d htz-e htz-f"
    id={null}
  >
    error text
  </span>
</div>
`;

exports[`<Note> DOM element renders correctly with \`isError prop 2`] = `".htz-a{display:block}.htz-f{color:#A8001C}@media (max-width: 79.9375em){.htz-b{font-size:1.8333333333333333rem}.htz-c{line-height:1.6363636363636365em}}@media (min-width: 80em){.htz-d{font-size:1.7142857142857142rem}.htz-e{line-height:1.75em}}"`;

exports[`<Note> DOM element renders correctly with minimal required props 1`] = `
<div>
  <span
    className="htz-a htz-b htz-c htz-d htz-e htz-f"
    id={null}
  >
    text
  </span>
</div>
`;

exports[`<Note> DOM element renders correctly with minimal required props 2`] = `".htz-a{display:block}.htz-f{color:#787878}@media (max-width: 79.9375em){.htz-b{font-size:1.8333333333333333rem}.htz-c{line-height:1.6363636363636365em}}@media (min-width: 80em){.htz-d{font-size:1.7142857142857142rem}.htz-e{line-height:1.75em}}"`;
